reactive/vpe_router.py

```diff
@@ -96,6 +96,101 @@ def add_corporation():
               iface_vlanid)
 
 
+@when('vpe.delete-corporation')
+def delete_corporation():
+
+    domain_name = action_get('domain-name')
+
+    # Remove all tunnels defined for this domain
+    p = router._run([
+        'ip',
+        'netns',
+        'exec',
+        'domain_name',
+        'ip',
+        'tun',
+        'show',
+        '|',
+        'grep',
+        'gre',
+        '|',
+        'grep',
+        '-v',
+        '"remote any"',
+        '|',
+        'cut -d":" -f1'
+    ])
+
+    # `p` should be a tuple of (stdout, stderr)
+    tunnels = p[0]
+
+    for tunnel in tunnels:
+        router.ip(
+            'netns',
+            'exec',
+            domain_name,
+            'ip',
+            'link',
+            'set',
+            tunnel,
+            'down'
+        )
+
+        router.ip(
+            'netns',
+            'exec',
+            domain_name,
+            'ip',
+            'tunnel',
+            'del',
+            tunnel
+        )
+
+    # Remove all interfaces associated to the domain
+    p = router._run(
+        'ip',
+        'netns',
+        'exec',
+        'domain_name',
+        'ifconfig',
+        '|',
```

Comment:
`shell != true` in subprocess, so this won't work. you'll either need to add that (don't) or do post processing in pyhton